hdk-rust/src/meta.rs

```diff
@@ -0,0 +1,58 @@
+use entry_definition::ValidatingEntryType;
+use holochain_wasm_utils::{error::RibosomeErrorCode, memory_serialization::load_string};
+
+trait Ribosome {
+    fn define_entry_type(&mut self, name: String, entry_type: ValidatingEntryType);
+}
+
+pub struct ZomeDefinition {
+    pub entry_types: Vec<ValidatingEntryType>,
+}
+
+impl ZomeDefinition {
+    fn new() -> ZomeDefinition {
+        ZomeDefinition {
+            entry_types: Vec::new(),
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn define(&mut self, entry_type: ValidatingEntryType) {
+        self.entry_types.push(entry_type);
+    }
+}
+
+extern "C" {
+    #[allow(improper_ctypes)]
+    fn zome_setup(zd: &mut ZomeDefinition);
+}
+
+#[no_mangle]
+pub extern "C" fn validation_package(encoded_allocation_of_input: u32) -> u32 {
+    ::global_fns::init_global_memory(encoded_allocation_of_input);
+
+    let mut zd = ZomeDefinition::new();
+    unsafe {
+        zome_setup(&mut zd);
+    }
+
+    // Deserialize input
+    let maybe_name = load_string(encoded_allocation_of_input);
+    if let Err(_) = maybe_name {
+        return ::holochain_wasm_utils::error::RibosomeErrorCode::ArgumentDeserializationFailed
```

Comment:
Seems I need to change load_string() to return a Result<String, RibosomeErrorCode> instead, since this is hiding the original RibosomeReturnCode in favor of ArgumentDeserializationFailed.

Reply:
Also:

```rust
if maybe_name.is_err() {
```

Reply:
👍